substrate/client/network/src/peer_info.rs

```diff
@@ -414,7 +414,8 @@ impl NetworkBehaviour for PeerInfoBehaviour {
 			},
 			FromSwarm::NewListenAddr(e) => {
 				self.ping.on_swarm_event(FromSwarm::NewListenAddr(e));
-				self.identify.on_swarm_event(FromSwarm::NewListenAddr(e));
+				// Do not report listen addresses to `Identify`, because remote nodes should only
+				// know about our external addresses.
```

Comment:
Not sure if this is a good idea. Probably needs a burn-in in a real network to see if it's safe to remove because AFAIU many of our nodes rely on external address discovery via Identify and the issues we've experienced are limited Rococo. If this is necessary to fix it, then it probably should be behind a CLI flag to enable this behavior explicitly.

Reply:
Yes, let's do a burn-in. This should not affect the external address discovery via Identify, as external addresses are handled separately via [`observed_addr`](https://github.com/libp2p/rust-libp2p/blob/v0.51/protocols/identify/src/handler.rs#L251) filed of Identify message.

Reply:
Yeah I was mixing up listen and external addresses

Reply:
The confusion is no surprise, as there is only one field `listen_addrs` in Identify message, where both listen and external addresses are added.